src/highlight.js

```diff
@@ -314,15 +319,19 @@ https://highlightjs.org/
   }
 
   /*
-  Core highlighting function. Accepts a language name, or an alias, and a
-  string with the code to highlight. Returns an object with the following
-  properties:
+  Core highlighting function. Accepts:
+
+  - name: a language name, or an alias
+  - value: a string with the code to highlight
+  - soft_illegals: if false will use strict parsing where illegals stop processing, if true will use illegals as an alternate end regex
+  - continuation:
+
+  Returns an object with the following properties:
 
   - relevance (int)
   - value (an HTML string with highlighting markup)
-
   */
-  function highlight(name, value, ignore_illegals, continuation) {
+  function highlight(name, value, soft_illegals, continuation) {
```

Comment:
so, instead of changing the function signature in this manner, I suggest you instead change it to accept an an object as the last argument:

``` JS
function highlight(name, value, ignore_illegals, continuation, options) {
}
``` 

Then pass the `options` to `compileLanguage`:

```JS
compileLanguage(language, options);
``` 

And then check the `options.soft_illegals`... I would also change  `soft_illegals` to `treat_illegals_as: "soft"`, that way, it's more extensible in the future.

Reply:
Definitely agree.